pkg/common/plugin/sshpop/sshpop.go

```diff
@@ -58,9 +57,6 @@ type Server struct {
 type ClientConfig struct {
 	HostKeyPath  string `hcl:"host_key_path"`
 	HostCertPath string `hcl:"host_cert_path"`
-	// CanonicalDomain specifies the domain suffix for validating the hostname against
-	// the certificate's valid principals. See CanonicalDomains in ssh_config(5).
-	CanonicalDomain string `hcl:"canonical_domain"`
```

Comment:
This was forgotten in #922

Reply:
If it's unused, we're essentially just deleting dead code. Even if the configuration existed, removing it doesn't change the behavior of the agent, so it won't have an impact on users. In other words, whether this config flag is set is a no-op both before and after we delete this code right? If so, I don't see a need to do the deprecation path - we could just remove it outright.

Reply:
Will removing this config parameter cause consumers who are setting it to receive errors on startup? Or are unused config parameters ignored?

Reply:
Only the configuration that is inspected by SPIRE core currently errors for unknown config values (and a subset at that due to a bug in HCL parsing). Plugins would have to add support individually. The sshpop plugin does not have it.

Reply:
:+1: I think it's safe to skip the deprecation path in this case